front/screens/search/tabAroundMeInstruction.component.tsx

```diff
@@ -59,18 +59,17 @@ const TabAroundMeInstruction: FC<Props> = ({ articles }) => {
 
   const extractPoiTypesFromArticles = () => {
     const finalCartographieTypes: PoiType[] = [];
-    articles
-      .map((article) => article.cartographie_pois_types)
-      .map((types) =>
-        types?.map((type) => {
-          if (
-            !finalCartographieTypes
-              .map((finalType) => finalType.nom)
-              .includes(type.nom)
-          )
-            finalCartographieTypes.push(type);
-        })
+    articles.map((article) => {
+      const pois = article.cartographie_pois_types?.filter(
+        (type) =>
+          !finalCartographieTypes
+            .map((finalType) => finalType.nom)
+            .includes(type.nom)
       );
+      if (pois && pois.length > 0)
+        pois.map((poi) => finalCartographieTypes.push(poi));
+    });
+
```

Comment:
Au final, tout ce code un peu obscure, ça ressemble à un `flatMap` : on construit un tableau à partir de tableaux (vide ou non, en fonction de l'itération).